src/core/agent/services/identifierService.ts

```diff
@@ -128,18 +127,33 @@ class IdentifierService extends AgentService {
     const operation = await this.signifyClient
       .identifiers()
       .create(signifyName); //, this.getCreateAidOptions());
-    await operation.op();
+    let op = await operation.op();
+    const signifyOpName = op.name;
     const addRoleOperation = await this.signifyClient
```

Comment:
We are always able to add the end role before `op` is done?

Reply:
Yep

Reply:
@iFergal Will I add `    let op = await operation.op(); ` after  add the end role ?

Reply:
@bao-sotatek The operation will still "complete" itself on KERIA as soon as possible no matter where we call `await operation.op()`. So I'm just wondering if we can always add the end role straight away - but I think maybe we'd need to inspect Signify and KERIA code or KERIA tests to see - lets cover it in another story